connect/client.go

```diff
@@ -404,7 +431,45 @@ func (rs *restClient) GetFileContent(file *onepassword.File) ([]byte, error) {
 	if content, err := file.Content(); err == nil {
 		return content, nil
 	}
+	response, err := rs.retrieveDocumentContent(file)
+	content, err := readResponseBody(response, http.StatusOK)
+	if err != nil {
+		return nil, err
+	}
+	file.SetContent(content)
+	return content, nil
+}
+
+func (rs *restClient) DownloadFile(fileUUID, itemUUID, vaultUUID, targetDirectory string) (string, error) {
+	file, err := rs.GetFile(fileUUID, itemUUID, vaultUUID)
+	if err != nil {
+		return "", err
+	}
+	response, err := rs.retrieveDocumentContent(file)
+	if err != nil {
+		return "", err
+	}
+
+	path := filepath.Join(targetDirectory, filepath.Base(file.Name))
+	dst, err := os.Create(path)
+	if err != nil {
+		return "", err
+	}
+	err = os.Chmod(path, 0600)
+	if err != nil {
+		return "", err
+	}
+	defer dst.Close()
+	if _, err = io.Copy(dst, response.Body); err != nil {
+		return "", err
+	}
 
+	abs, err := filepath.Abs(path)
```

Comment:
Is there a specific reason for returning the _absolute_ path? We're now having to deal with this return error. 

What if we just returned the `path` value? Then we leave it up to the user what they want. If they pass an absolute path to `DownloadFile()`, they get an absolute path back. If they pass a relative path, they get back a relative path. Or we could also just return the filename and leave it up to the user to combine that with the `targetDirectory` they passed.

Reply:
I thought about returning the absolute path as that can also be used outside the scope of the working directory, in case the file needs to be referenced.

On a second thought, you are right, I fixed the function to return the joined path passed by the user, together with the filename.